bokeh/io.py

```diff
@@ -346,6 +356,47 @@ def _save_helper(obj, filename, resources, title, validate):
         with io.open(filename, "w", encoding="utf-8") as f:
             f.write(decode_utf8(html))
 
+def savepng(obj, filename=None, state=None, validate=True):
+    if state is None:
+        state = _state
+
+    if filename is None and state.png:
+        filename = state.png['filename']
+
+    if filename is None:
+        raise RuntimeError("savepng() called but no filename was supplied and output_png(...) was never called, nothing saved")
+
+    with _ModelInDocument(obj):
+        if isinstance(obj, Component):
+            doc = obj.document
+        elif isinstance(obj, Document):
+            doc = obj
+        else:
+            raise RuntimeError("Unable to save object of type '%s'" % type(obj))
+
+        if validate:
+            doc.validate()
+
+        from subprocess import Popen, PIPE
+        from os.path import join, dirname
+
+        from .resources import Resources
+
+        resources = Resources(mode="absolute-dev")
+
+        from .embed import _check_plot_objects, _standalone_docs_json_and_render_items
+        from ._json_encoder import serialize_json
+
+        plot_objects = [obj]
+        plot_objects = _check_plot_objects(plot_objects)
+        (docs_json, render_items) = _standalone_docs_json_and_render_items(plot_objects)
```

Comment:
These underscore-prefixed functions aren't really meant to leave `embed.py`; it might be nice to add an embed.py function specifically for this, `standalone_docs_json_for_png(plot_objects)` which would essentially contain the above three lines.